spiffworkflow-frontend/src/routes/TaskShow.tsx

```diff
@@ -39,48 +39,55 @@ export default function TaskShow() {
     permissionRequestData
   );
 
+  const processResult = (result: any) => {
+    setTask(result);
+    const url = `/task-data/${modifyProcessIdentifierForPathParam(
+      result.process_model_identifier
+    )}/${params.process_instance_id}`;
+    if (
+      result.process_model_identifier &&
+      ability.can('GET', url)
+      // Assure we get a valid process model identifier back
+    ) {
+      HttpService.makeCallToBackend({
+        path: url,
+        successCallback: setUserTasks,
+        failureCallback: (error: any) => {
+          addError(error);
+        },
+      });
+    }
+  };
+
   useEffect(() => {
     if (permissionsLoaded) {
-      const processResult = (result: any) => {
-        setTask(result);
-        if (ability.can('GET', targetUris.processInstanceTaskListDataPath)) {
-          HttpService.makeCallToBackend({
-            path: `/task-data/${modifyProcessIdentifierForPathParam(
-              result.process_model_identifier
-            )}/${params.process_instance_id}`,
-            successCallback: setUserTasks,
-          });
-        }
-      };
-
       HttpService.makeCallToBackend({
         path: `/tasks/${params.process_instance_id}/${params.task_id}`,
         successCallback: processResult,
-        // This causes the page to continuously reload
-        // failureCallback: setErrorObject,
+        failureCallback: addError,
       });
     }
-  }, [params, permissionsLoaded, ability, targetUris]);
+  }, [permissionsLoaded, ability]); // params and targetUris (which deps on params) cause this to re-fire in an infinite loop on error.
```

Comment:
I don't think the issue is with the params and targetUris. It may have to do with addError though.